### PR TITLE
Clean up null issues, add more annotations

### PR DESCRIPTION
**Pull request by mezz**
This is a larger static analysis for null issues in Forge.
I tried to go over everything to find any `ItemStack` that was being handled incorrectly, as well as other minor null handling issues.

I think this should only be pulled after the next round of PRs, it will probably conflict with a lot of stuff.

### CODE REVIEW

File: patches/minecraft/net/minecraft/world/chunk/storage/RegionFile.java.patch [hunk 1]
```diff
@@ -37,6 +37,6 @@
 +        return false;
 +    }
 +
+     @Nullable
+ 
      public synchronized DataInputStream func_76704_a(int p_76704_1_, int p_76704_2_)
-     {
-         if (this.func_76705_d(p_76704_1_, p_76704_2_))
```
Comment (mezz) [inline patches/minecraft/net/minecraft/world/chunk/storage/RegionFile.java.patch:42]: In this file, the earlier patch was applied in the wrong spot. The `@Nullable` was supposed to go above `func_76704_a` (weirdly with a new line between).
This patch puts it in the right place.

**Review by liach — APPROVED** [review]

**Comment by mezz** [comment]
Closing and opening to trigger a Travis build.

**Comment by mezz** [comment]
`extractForgeRangemap` took so long without printing anything (>10 minutes) that Travis timed out. D'oh.

**Comment by williewillus** [comment]
can we cache that stuff somehow?

**Comment by mezz** [comment]
I barely know what 90% of the ForgeGradle steps do, I'll have to look into it more.

**Comment by diesieben07** [comment]
There is a way to make travis keep the gradle cache in between builds:
```
cache:
  directories:
    - $HOME/.gradle
```

**Comment by AEnterprise** [comment]
you can just tell travis to wait longer with ```travis_wait``` and how many minutes to wait

**Comment by williewillus** [comment]
+1 for getting this in, it's causing crashes since e.g. RenderTooltipEvent.PostText is currently firing with null stacks still even though it's annotated nonnull

**Comment by mezz** [comment]
@diesieben07 I already cache 
```
cache:
  directories:
    - $HOME/.gradle/caches/
    - $HOME/.gradle/wrapper/
```
I will look into this more though, my instance may be completely new each time depending on other settings.
I'm not sure what the inputs and outputs are for `extractForgeRangemap` so I'm not sure if it is something that can be cached or not.

@AEnterprise  `travis_wait` should work for now, thanks.

**Pull request merged**